.github/workflows/unit-test.yml

```diff
@@ -7,7 +7,7 @@ jobs:
     runs-on: ubuntu-20.04
     strategy:
       matrix:
-        node: [10, 12, 14]
+        node: [18, 20, 22]
```

Comment:
If we are still supporting Node v10 & friends we should still test on them

If the problem is only the lock file, can we delete it instead of removing them from the matrix?

Reply:
Given that those node versions are out of support by the maintainers I'm wondering if we necessarily need to support them?

I've tried removing the package lock and adding back these node versions anyway.

Reply:
@imjoehaines it appears that the tests don't run on those unsupported node versions because of a missing dependency on `node:fs/promises`.

Reply:
With the latest changes, we're now supporting back to the earliest Node we can whilst still avoiding the vulnerabilities reported, albeit only in dev dependencies. Given the age of the dropped Node versions in test and that this is a build tool and not used in the SDK, I feel this is defensible.

@imjoehaines - thoughts?